### PR TITLE
Only check the sharing backends for file/folder items

### PR DESCRIPTION
**Pull request by nickvergessen**
Required for https://github.com/nextcloud/spreed/issues/67#issuecomment-259676033

@LukasReschke @rullzer

### CODE REVIEW

File: apps/files_sharing/lib/Controller/ShareesAPIController.php [hunk 1]
```diff
@@ -427,11 +427,25 @@ public function search($search = '', $itemType = null, $page = 1, $perPage = 200
...
 		$shareTypes = [
 			Share::SHARE_TYPE_USER,
-			Share::SHARE_TYPE_REMOTE,
-			Share::SHARE_TYPE_EMAIL
+			Share::SHARE_TYPE_EMAIL,
 		];
 
-		if ($this->shareManager->allowGroupSharing()) {
+		if ($itemType === 'file' || $itemType === 'folder') {
+			$shareTypes[] = Share::SHARE_TYPE_REMOTE;
```
Comment (rullzer) [inline apps/files_sharing/lib/Controller/ShareesAPIController.php]: Why remove it above to add it here in the if and the else statement?

**Review by rullzer — CHANGES_REQUESTED** [review]

File: apps/files_sharing/lib/Controller/ShareesAPIController.php (hunk 1 above)
Comment (rullzer) [inline apps/files_sharing/lib/Controller/ShareesAPIController.php]: Ah pfff stupid me sorry.

**Review by rullzer — APPROVED** [review]

**Comment by rullzer** [comment]
If CI is fine I'm fine :) :+1:

File: apps/files_sharing/lib/Controller/ShareesAPIController.php [hunk 2]
```diff
@@ -427,11 +427,25 @@ public function search($search = '', $itemType = null, $page = 1, $perPage = 200
...
+			if ($this->shareManager->allowGroupSharing()) {
+				$shareTypes[] = Share::SHARE_TYPE_GROUP;
+			}
+
+			if (in_array(Share::SHARE_TYPE_REMOTE, $shareTypes) && !$this->isRemoteSharingAllowed($itemType)) {
```
Comment (MorrisJobke) [inline apps/files_sharing/lib/Controller/ShareesAPIController.php]: This statement looks silly - could we combine this with line 434 to only add the remote share type if remote sharing is allowed?

**Review by MorrisJobke — CHANGES_REQUESTED** [review]

File: apps/files_sharing/lib/Controller/ShareesAPIController.php (hunk 2 above)
Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/ShareesAPIController.php]: just copy pasted, but I guess we could

Comment (MorrisJobke) [inline apps/files_sharing/lib/Controller/ShareesAPIController.php]: I know that you copy pasted, but then the logic is even more weird ;) So maybe see this as a way to enhance readability. ;)

Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/ShareesAPIController.php]: fixed

**Comment by MorrisJobke** [comment]
👍

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**